app/code/Magento/Rule/Model/Condition/AbstractCondition.php

```diff
@@ -886,6 +885,9 @@ public function validateAttribute($validatedValue)
     protected function _compareValues($validatedValue, $value, $strict = true)
     {
         if ($strict && is_numeric($validatedValue) && is_numeric($value)) {
+            if (preg_match('/^0\d+$/', $validatedValue) || preg_match('/^0\d+$/', $value)) {
```

Comment:
We can simplified the code while maintaining the same logic as below:
```suggestion
           $pattern = '/^0\d+$/';
        if ((preg_match($pattern, $validatedValue) || preg_match($pattern, $value))) {
```